app/components/shared/RouteTransition.js

```diff
@@ -0,0 +1,45 @@
+import { createElement as h } from "react";
+import { spring, TransitionMotion as TM } from "react-motion";
+import { object, string, func, bool } from "prop-types";
+
+const e = (s, o) => Object.keys(s).reduce((a, b) => { a[b] = typeof s[b] === "number" ? spring(s[b], o) : s[b]; return a; }, {});
```

Comment:
Can you add some explicit variable names, please? Not possible to understand what is doing without spending time on it

Reply:
sure.. it'll add about 15 lines of code to make this whole thing explicit.